File: app-android/app/src/test/java/de/tutao/tutanota/UtilsTest.java
package de.tutao.tutanota;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class UtilsTest {
	@Test
	public void testIsColorLightPinkDark() {
		assertFalse(Utils.isColorLight("#B73A9A"));
	}

	@Test
	public void testIsColorLightBlueLight() {
		assertTrue(Utils.isColorLight("#3A9AFF"));
	}

	@Test
	public void testIsThreeDigitBlackDark() {
		assertFalse(Utils.isColorLight("#000"));
	}

	@Test
	public void testIsThreeDigitWhiteLight() {
		assertTrue(Utils.isColorLight("#FFF"));
	}

	@Test
	public void testIsThreeDigitCyanLight() {
		assertTrue(Utils.isColorLight("#0FF"));
	}

	@Test
	public void testParseSixDigitColors() {
		assertEquals(Utils.parseColor("#000000"), 0xff000000);
		assertEquals(Utils.parseColor("#FFFFFF"), 0xffffffff);
		assertEquals(Utils.parseColor("#FABDAD"), 0xfffabdad);
	}

	@Test
	public void testParseThreeDigitColors() {
		assertEquals(Utils.parseColor("#000"), 0xff000000);
		assertEquals(Utils.parseColor("#FFF"), 0xffffffff);
		assertEquals(Utils.parseColor("#BAD"), 0xffbbaadd);
	}

}

File: app-android/calendar/src/test/java/de/tutao/calendar/UtilsTest.kt
package de.tutao.calendar

import de.tutao.tutashared.isAllDayEventByTimes
import de.tutao.tutashared.isLightHexColor
import de.tutao.tutashared.parseColor
import org.junit.Assert
import org.junit.Test
import java.time.Instant
import java.time.LocalDateTime
import java.util.Date

class UtilsTest {
	@Test
	fun testIsColorLightPinkDark() {
		Assert.assertFalse("#B73A9A".isLightHexColor())
	}

	@Test
	fun testIsColorLightBlueLight() {
		Assert.assertTrue("#3A9AFF".isLightHexColor())
	}

	@Test
	fun testIsThreeDigitBlackDark() {
		Assert.assertFalse("#000".isLightHexColor())
	}

	@Test
	fun testIsThreeDigitWhiteLight() {
		Assert.assertTrue("#FFF".isLightHexColor())
	}

	@Test
	fun testIsThreeDigitCyanLight() {
		Assert.assertTrue("#0FF".isLightHexColor())
	}

	@Test
	fun testParseSixDigitColors() {
		Assert.assertEquals(parseColor("#000000").toLong(), -0x1000000)
		Assert.assertEquals(parseColor("#FFFFFF").toLong(), -0x1)
		Assert.assertEquals(parseColor("#FABDAD").toLong(), -0x54253)
	}

	@Test
	fun testParseThreeDigitColors() {
		Assert.assertEquals(parseColor("#000").toLong(), -0x1000000)
		Assert.assertEquals(parseColor("#FFF").toLong(), -0x1)
		Assert.assertEquals(parseColor("#BAD").toLong(), -0x445523)
	}

	@Test
	fun testIsAllDayByTimesFunction() {
		val dateStart = Date.from(Instant.ofEpochMilli(1758326400000L))
		val dateEnd = Date.from(Instant.ofEpochMilli(1758412800000L))
		Assert.assertTrue(isAllDayEventByTimes(dateStart, dateEnd))
	}

	@Test
	fun testIsNotAllDayByTimesFunction() {
		val dateStart = Date.from(Instant.ofEpochMilli(1758326400000L))
		val dateEnd = Date.from(Instant.ofEpochMilli(1758319200000L))
		Assert.assertFalse(isAllDayEventByTimes(dateStart, dateEnd))
	}
}